jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt

```diff
@@ -91,20 +108,35 @@ class LocalLambdaRunConfiguration(project: Project, factory: ConfigurationFactor
                 ShowSettingsUtil.getInstance().showSettingsDialog(project, AwsSettingsConfigurable::class.java)
             }
         }
+    }
 
-        resolveCredentials()
+    private fun checkLambdaHandler() {
+        val handlerValidator = project.service<LambdaHandlerValidator>()
+        val (handler, runtime) = resolveLambdaInfo(project = project, functionOptions = lambdaOptions.functionOptions)
+        val promise = handlerValidator.evaluate(LambdaHandlerValidator.LambdaEntry(project, runtime, handler))
+
+        if (promise.isPending) {
+            promise.then { isValid ->
+                messageBus.syncPublisher(
+                    LambdaHandlerEvaluationListener.TOPIC).handlerValidationFinished(handler, isValid)
+            }
+
+            logger.info { "Validation will proceed asynchronously for SAM CLI version" }
```

Comment:
Please change the log message here: here we validate the lambda handler, not the SAM CLI version.